apps/ledger-live-desktop/src/renderer/components/WebPlatformPlayer/WebViewV2.tsx

```diff
@@ -485,31 +485,44 @@ export function WebView({ manifest, onClose, inputs = {}, config }: Props) {
     };
   }, [manifest, handleMessage]);
 
-  const handleNewWindow = useCallback(async e => {
-    const protocol = new URL(e.url).protocol;
-    if (protocol === "http:" || protocol === "https:") {
-      await shell.openExternal(e.url);
+  const handleDomReady = useCallback(() => {
+    const webview = targetRef.current;
+    if (!webview) {
+      return;
     }
+
+    const id = webview.getWebContentsId();
+
+    // cf. https://gist.github.com/codebytere/409738fcb7b774387b5287db2ead2ccb
+    // @ts-expect-error: missing typings for api
+    window.api.openWindow(id);
   }, []);
 
   useEffect(() => {
     const webview = targetRef.current;
 
     if (webview) {
-      // For mysterious reasons, the webpreferences attribute does not
-      // pass through the styled component when added in the JSX.
-      webview.webpreferences = "nativeWindowOpen=no";
-      webview.addEventListener("new-window", handleNewWindow);
       webview.addEventListener("did-finish-load", handleLoad);
+      webview.addEventListener("dom-ready", handleDomReady);
     }
 
     return () => {
       if (webview) {
-        webview.removeEventListener("new-window", handleNewWindow);
         webview.removeEventListener("did-finish-load", handleLoad);
+        webview.removeEventListener("dom-ready", handleDomReady);
       }
     };
-  }, [handleLoad, handleNewWindow]);
+  }, [handleLoad, handleDomReady]);
+
+  const webviewStyle = useMemo(() => {
+    return {
+      opacity: widgetLoaded ? 1 : 0,
+      border: "none",
+      width: "100%",
+      flex: 1,
+      transition: "opacity 200ms ease-out",
+    };
+  }, [widgetLoaded]);
```

Comment:
I lack context as to why we have two versions of this.
Can't we kill one of them already?

Reply:
it's a good point but this is unrelated to Electron 22, we just had to make sure both works still properly. ( cc @Justkant  )
this feedback need to be taken somewhere else (unrelated to this PR to "kill one of them")

Reply:
We just introduced `WebViewV2` in order to support the new wallet-api while not touching the old platform stuff for the `live-app-sdk`